linera-client/src/benchmark.rs

```diff
@@ -737,3 +728,38 @@ impl<Env: Environment> Benchmark<Env> {
         }
     }
 }
+
+struct ChainDestinationManager {
+    destination_index: usize,
+    num_destinations: usize,
+    destination_chains: Vec<ChainId>,
+    rng: SmallRng,
+}
+
+impl ChainDestinationManager {
+    fn new(mut destination_chains: Vec<ChainId>) -> Result<Self, BenchmarkError> {
+        let mut rng = SmallRng::from_rng(thread_rng())?;
+        destination_chains.shuffle(&mut rng);
+
+        Ok(Self {
+            destination_index: 0,
+            num_destinations: destination_chains.len(),
+            destination_chains,
+            rng,
+        })
+    }
+
+    fn get_next_destination(&mut self) -> ChainId {
+        // Check if we've gone through all destinations
+        if self.destination_index >= self.num_destinations {
+            // Reshuffle the destinations for the next cycle
```

Comment:
Do we need that? Otherwise the whole destination manager could be replaced by an iterator using `cycle`.